qa/vagrant/src/main/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -61,4 +90,220 @@ public void test10Install() {
+                "$acl.SetOwner($account); " +
+                "Set-Acl '" + installation.config("elasticsearch.keystore") + "' $acl"
+            );
+        } else {
+            sh.bash("sudo -u " + ARCHIVE_OWNER + " " + installation.bin("elasticsearch-keystore") + " create");
+        }
+
+        assertThat(installation.config("elasticsearch.keystore"), file(File, ARCHIVE_OWNER, ARCHIVE_OWNER, p660));
+
+        final Result r = Platforms.WINDOWS
+            ? sh.powershell(installation.bin("elasticsearch-keystore.bat") + " list")
+            : sh.bash("sudo -u " + ARCHIVE_OWNER + " " + installation.bin("elasticsearch-keystore") + " list");
+        assertThat(r.stdout, containsString("keystore.seed"));
+
+        // cleanup for next test
+        rm(installation.config("elasticsearch.keystore"));
+    }
+
+    @Test
+    public void test50StartAndStop() {
+        assumeThat(installation, is(notNullValue()));
+
+        Archives.runElasticsearch(installation);
+
+        final String gcLogName = Platforms.LINUX
+            ? "gc.log.0.current"
+            : "gc.log";
+        assertTrue("gc logs exist", Files.exists(installation.logs.resolve(gcLogName)));
+        ServerUtils.runElasticsearchTests();
+
+        Archives.stopElasticsearch(installation);
+    }
+
+    @Test
+    public void test60AutoCreateKeystore() {
+        assumeThat(installation, is(notNullValue()));
+
+        assertThat(installation.config("elasticsearch.keystore"), file(File, ARCHIVE_OWNER, ARCHIVE_OWNER, p660));
+
+        final Shell sh = new Shell();
+        final Result result;
+        if (Platforms.WINDOWS) {
+            result = sh.powershell(installation.bin("elasticsearch-keystore.bat") + " list");
+        } else {
+            result = sh.bash("sudo -u " + ARCHIVE_OWNER + " " + installation.bin("elasticsearch-keystore") + " list");
+        }
+        assertThat(result.stdout, containsString("keystore.seed"));
+    }
+
+    @Test
+    public void test70CustomPathConfAndJvmOptions() {
+        assumeThat(installation, is(notNullValue()));
+
+        final Path tempConf = getTempDir().resolve("esconf-alternate");
+
+        try {
+            mkdir(tempConf);
+            cp(installation.config("elasticsearch.yml"), tempConf.resolve("elasticsearch.yml"));
+            cp(installation.config("log4j2.properties"), tempConf.resolve("log4j2.properties"));
+
+            // we have to disable Log4j from using JMX lest it will hit a security
+            // manager exception before we have configured logging; this will fail
+            // startup since we detect usages of logging before it is configured
+            final String jvmOptions =
+                "-Xms512m\n" +
+                "-Xmx512m\n" +
+                "-Dlog4j2.disable.jmx=true\n";
+            append(tempConf.resolve("jvm.options"), jvmOptions);
+
+            final Shell sh = new Shell();
+            if (Platforms.WINDOWS) {
+                sh.powershell(
```

Comment:
I'm not sure we can change these permissions on windows in java because of how it exposes filesystem attributes for windows' fs. Or at least that was what I originally intended to do but couldn't get it working. Is there something you could point me towards?

But yeah definitely right about the user not doing it this way, they'll use the gui